qxf2/newsletter_automation
# Accessibility Tests for Newsletter

This PR is for accessibility tests for newsletter.

I have added a new method in the page factory to return all the pages. The pages are being iterated and injected axe-core and run for checking accessibility.

Once the accessibility checks are run, a snapshot is created for every page.

So, next time a new page is introduced, we just need to add it to the page factory, and the accessibility script will do the job.

An accessibility util is added which has the methods to inject, run, and write axe into the tests.

Steps to run the test:
1. Go to the accessibility _tests folder
2. Run the test using python -m pytest test_accessibility.py --snapshot-update (We use --snapshot-update only for the first time to create the snapshot directory)
3. Once the test is run, you can see the accessibility for each page inside the snapshot folder.

<img width="958" alt="output" src="https://github.com/qxf2/newsletter_automation/assets/97028279/bea473a5-89e9-45e0-b159-a194f3a127b2">
